### DISCUSSION TITLE
Use rich's default repr for user classes

### DISCUSSION THREAD

**Discussion by sylvorg** [Textualize/rich#3267] [Q&A]
Hello!

Sorry, I'm having trouble explaining exactly what I want, but is there any way for a class not inheriting from, say, a dict, to have the repr of the dict? For example, when pretty printing a dictionary, larger dictionaries are expanded by default, but doing something like `self.__repr__ = self.dct.__repr__` doesn't seem work, and the built-in dict type obviously has no `__rich_repr__` for me to copy. Is there any way to get the same default expansion for user classes?

Thank you kindly for the help!

**Comment by TomJGooding** [comment]
I'm not sure I fully understand your question, but you might find the [Rich Repr Protocol](https://rich.readthedocs.io/en/stable/pretty.html?highlight=repr#rich-repr-protocol) section helpful in the docs about Pretty Printing.

**Reply by sylvorg** [reply]
Sorry, copying my original reply to you from below:

I use that regularly, but my question is how to use the built-in formatting, for situations such as the following:

```python
class Test:
    def __init__(self, **kwargs):
        self.kwargs = kwargs
        self.__repr__ = self.kwargs.__repr__
```

Rich doesn't seem to use the repr for the built-in type. Is there any way to achieve this sort of functionality?

**Comment by sylvorg** [comment]
I use that regularly, but my question is how to use the built-in formatting, for situations such as the following:

```python
class Test:
    def __init__(self, **kwargs):
        self.kwargs = kwargs
        self.__repr__ = self.kwargs.__repr__
```

Rich doesn't seem to use the repr for the built-in type. Is there any way to achieve this sort of functionality?

**Reply by willmcgugan** [reply]
Not as such. The closest you will get is yielding a dict from `__rich_repr__`. That will wrap the dict like `MyDictLike({"foo": "bar"})`

**Reply by sylvorg** [reply]
Ah; I was under the impression you needed to always yield a string. I'll try this out and report back!

**Reply by sylvorg** [reply]
It works, @willmcgugan! However, is there any way to remove the class name and parentheses?
